vivisect/analysis/amd64/emulation.py

```diff
@@ -33,96 +31,62 @@ def prehook(self, emu, op, starteip):
             if len(op.opers):
                 self.retbytes = op.opers[0].imm
 
-##### FIXME: this should be all done through the calling convention
-sysvamd64argnames = {
-    0: ('rdi', e_amd64.REG_RDI),
-    1: ('rsi', e_amd64.REG_RSI),
-    2: ('rdx', e_amd64.REG_RDX),
-    3: ('rcx', e_amd64.REG_RCX),
-    4: ('r8',  e_amd64.REG_R8),
-    5: ('r9',  e_amd64.REG_R9),
-}
-
-msx64argnames = {
-    0: ('rcx', e_amd64.REG_RCX),
-    1: ('rdx', e_amd64.REG_RDX),
-    2: ('r8',  e_amd64.REG_R8),
-    3: ('r9',  e_amd64.REG_R9),
-}
-
-arch_bindings = {
-    'msx64call': msx64argnames,
-    'sysvamd64call': sysvamd64argnames,
-    None: [],
-}
-
-def sysvamd64name(idx):
-    ret = sysvamd64argnames.get(idx)
-    if ret == None:
-        name = 'arg%d' % idx
-    else:
-        name, idx = ret
-    return name
-
-def msx64name(idx):
-    ret = msx64argnames.get(idx)
-    if ret == None:
-        name = 'arg%d' % idx
-    else:
-        name, idx = ret
-    return name
-#####
-
```

Comment:
this approach will become more and more cumbersome as we add more architectures.  i don't want to add arm and 8051 stuff here as well.